DTSPB-4825 notify when move to post grant issued

tom-saunders-cts opened the pull request:
### JIRA link (if applicable) ###
See [DTSPB-4825](https://tools.hmcts.net/jira/browse/DTSPB-4825)


### Change description ###



**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

tom-saunders-cts committed:
skeleton of notif behaviour for post grant issue.

tom-saunders-cts committed:
add handling for new notification template(s?).

tom-saunders-cts committed:
skeleton for sending notifications, doesn't store.

tom-saunders-cts committed:
now should store responses.

also adds handling for welsh dates, and begins to prep tests.

tom-saunders-cts committed:
Further tests, updates to resolve issues found with tests.

tom-saunders-cts committed:
don't send email to caseworker on failure.

tom-saunders-cts committed:
some sonar changes.

tom-saunders-cts committed:
remove unneeded secUtils.

FeliTam commented on src/main/java/uk/gov/hmcts/probate/service/NotificationService.java:
```diff
@@ -973,4 +980,66 @@ private List<String> getExecutorsNamesList(List<CollectionMember<ExecutorApplyin
+        } catch (NotificationClientException e) {
+            log.info("Failed to send Post Grant Issued notification for case {}, message: {}", caseRef, e.getMessage());
+            return null;
+        }
```
if the email was sent successfully but failed to upload, or an error occurred during **getGeneratedSentEmailDocument**, the exception is not handled here.

tom-saunders-cts replied:
i've updated this to explicitly catch and log any runtimeexception in the document generation / upload process

tom-saunders-cts committed:
be explicit about ignoring exceptions after the email has sent.

this means the event succeeds, but we might want to check that we aren't ignoring 'important' failures.

tom-saunders-cts committed:
remove unneeded log re: caseworker info.

tom-saunders-cts committed:
add missing coverage

tom-saunders-cts committed:
Merge branch 'master' into DTSPB-4825-notify-when-move-to-post-grant-issued

tom-saunders-cts committed:
accidentlaly duplicated member in merge.

tom-saunders-cts committed:
remove defunct test.

this was originally covering a case where the caaseworker was emailed
if there were some issue sending the notification, but that behaviour
was removed, so this test is not needed.
